File: backend/src/test/java/com/example/upload/standard/util/FileUtTest.java
package com.example.upload.standard.util;

import com.example.upload.global.app.AppConfig;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
@ActiveProfiles("test")
public class FileUtTest {

    @Test
    @DisplayName("downloadByHttp")
    public void t1() {
        String newFilePath = Ut.file.downloadByHttp("https://picsum.photos/id/237/200/300", AppConfig.getTempDirPath());

        // newFilePath 의 확장자가 jpg 인지 확인
        assertThat(newFilePath).endsWith(".jpg");

        Ut.file.delete(newFilePath);
    }

    @Test
    @DisplayName("getExtensionByTika")
    void tika() {
        String newFilePath = Ut.file.downloadByHttp("https://picsum.photos/id/237/200/300", AppConfig.getTempDirPath());

        String ext = Ut.file.getExtensionByTika(newFilePath);
        assertThat(ext).isEqualTo("jpg");

        Ut.file.delete(newFilePath);
    }
}

File: backend/src/test/java/com/example/upload/standard/util/FileUtTest.kt
package com.example.upload.standard.util

import com.example.upload.global.app.AppConfig.Companion.getTempDirPath
import com.example.upload.standard.util.Ut.file.delete
import com.example.upload.standard.util.Ut.file.downloadByHttp
import com.example.upload.standard.util.Ut.file.getExtensionByTika
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles

@SpringBootTest
@ActiveProfiles("test")
class FileUtTest {
    @Test
    @DisplayName("downloadByHttp")
    fun t1() {
        val newFilePath = downloadByHttp("https://picsum.photos/id/237/200/300", getTempDirPath())

        // newFilePath 의 확장자가 jpg 인지 확인
        Assertions.assertThat(newFilePath).endsWith(".jpg")

        delete(newFilePath)
    }

    @Test
    @DisplayName("getExtensionByTika")
    fun tika() {
        val newFilePath = downloadByHttp("https://picsum.photos/id/237/200/300", getTempDirPath())

        val ext = getExtensionByTika(newFilePath)
        Assertions.assertThat(ext).isEqualTo("jpg")

        delete(newFilePath)
    }
}